use ET_CHECK macro for sanity checks in memory shim layer

Gasoonjia opened the pull request:
Summary: this is a comprehensive update to use ET_CHECK macro to replace original if..else check for better follow et's law

Reviewed By: larryliu0820

Differential Revision: D83315545

larryliu0820 reviewed (approved):
Review automatically exported from Phabricator review in Meta.

Gasoonjia committed:
aoti_torch_copy_ (#14615)

Summary:

This diff introduce `aoti_torch_copy_`, the function for copying tensor inside cuda backend.

Right now it only support copy between tensors with same dtype.

Reviewed By: larryliu0820

Differential Revision: D83094604

Gasoonjia committed:
use ET_CHECK macro for sanity checks in memory shim layer (#14621)

Summary:

this is a comprehensive update to use ET_CHECK macro to replace original if..else check for better follow et's law

Reviewed By: larryliu0820

Differential Revision: D83315545

Gasoonjia committed:
aoti_torch_create_tensor_from_blob_v2 (#14604)

Summary:

This function introduce aoti_torch_create_tensor_from_blob_v2, a function that create tensor from data blob and custom stride and size.

Worth to notice that unlike aoti_torch_empty_strided, the tensor created by aoti_torch_create_tensor_from_blob_v2 will not have the control of the memory blob. Therefore when we delete it, the memory will not be freed.

Reviewed By: larryliu0820

Differential Revision: D83094602

Gasoonjia committed:
tensor destroy (#14575)

Summary:

This diff introduce `aoti_torch_delete_tensor_object` for deleting tensors created during cuda backend inference.

Reviewed By: larryliu0820

Differential Revision: D83094605

Gasoonjia committed:
tensor empty strided (#14549)

Summary:

this diff introduce aoti_tensor_empty_strided to et cuda backend, which will be one of the main functions to create empty tensor using the given stride.

Reviewed By: larryliu0820

Differential Revision: D83094606

Gasoonjia committed:
aoti_torch__reinterpret_tensor (#14614)

Summary:

Introduced aoti_torch__reinterpret_tensor, which creates a new tensor view that reinterprets the same underlying memory with custom shape and strides.

Reviewed By: larryliu0820

Differential Revision: D83094603
